src/core/engines/cross-margin/CrossMarginMath.sol

```diff
@@ -211,13 +239,23 @@ library CrossMarginMath {
         int256 intrinsicValue;
         int256[] memory weights;
 
+        // using put/call parity to convert puts to calls
         (strikes, weights, syntheticUnderlyingWeight, intrinsicValue) = _convertPutsToCalls(_detail);
 
+        // points-of-interest, array of strikes needed to evaluate collateral requirements
         pois = _createPois(strikes, _detail.putStrikes.length);
 
+        // payouts at each point-of-interest
         payouts = _calcPayouts(pois, strikes, weights, syntheticUnderlyingWeight, _detail.spotPrice, intrinsicValue);
     }
 
+    /**
+     * @notice generating points of interest (strikes)
+     * @dev adding a point left of left most strike (puts) and a right point of right most strike (call)
+     * @param strikes array of shorts and longs
+     * @param numOfPuts number of puts
+     * @return pois array of point-of-interests (aka strikes)
+     */
     function _createPois(uint256[] memory strikes, uint256 numOfPuts) internal pure returns (uint256[] memory pois) {
         uint256 epsilon = strikes.min() / 10;
```

Comment:
you should cache `strikes.min()` here, so you don't have to re-calculate it later.
Even better if you can use 1 loop to find min and max